### DISCUSSION TITLE
Having issues with loading the default values

### DISCUSSION THREAD

**Discussion by solecity** [react-hook-form/react-hook-form#8081] [Q&A]
I'm using inputs with controller component and material-ui inputs.

```
  const {
    control,
    handleSubmit,
    reset,
    formState: { isSubmitSuccessful, errors }
  } = useForm({
    mode: "onBlur",
    defaultValues: {
      name: account.name || "",
      type: getDefaultType(),
      balance: account.balance || 0,
      openingDate: account.openingDate || new Date(),
      colour: account.colour || DEFAULT.COLOUR,
      hasCard: account.hasCard || false,
      allocateGoal: account.allocateGoal || false
    },
    resolver: yupResolver(schemaAccount)
  });
```

In this example the account exists and has data but it never overrides the or option. What i need is when i click an edit button the form loads the data of that account but that does not happen. It worked before but now I don't know what I'm missing.

If I set the default values outside the useForm and pass as a variable then I can see those defaultValues are being updated but for some reason the useForm doesn't update.

**Comment by Moshyfawn** [comment]
Do you unmount your form component where the `useForm` is called in between the `defaultValues` change?

**Reply by solecity** [reply]
I'm not sure. How should I do that?

**Reply by Moshyfawn** [reply]
Maybe my question is out of place for your scenario. I was wondering if your form is a generic create / edit form that you might close _(by unmounting / not rendering the `Form` component)_ and so, whether or not the `useForm` hook is inside that component.

Maybe you check out [this CSB](https://codesandbox.io/s/tender-wind-tge3hp?file=/src/Page.tsx) I've created to roughly try re-creating your scenario. You can tweak it to closer match your use-case in which you've encountered your problem.

**Reply by bluebill1049** [reply]
are those props coming async or later on the stage? if that's the case you should reset the form at `useEffect`.

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api/useform

> defaultValues are cached on the first render within the custom hook. If you want to [reset](https://react-hook-form.com/api/useform/reset) the defaultValues, you should use the reset api.

**Reply by solecity** [reply]
thank you! that really helped

**Reply by bluebill1049** [reply]
np
